### ISSUE TITLE
Losing decimal point of numbers in table value parameter passed to Stored Procedure

### ISSUE THREAD

**Issue by milenkovyua** [TimelordUK/node-sqlserver-v8#156]
After the fix of issue #154 all the decimal values are now passed to the driver and the SQL server. During the work, I noticed another bug on transforming the decimal values (probably in the C++ part of the driver), before being sent to the SQL server. 

Here is the function I am calling:
```
    function callProcedureSimpleWithDecimalTVPFail(pool) {
        const tvpParam = tableTypes.get('testschema.withDecimalTypeTT');
        tvpParam.rows.add(30.11); // percentage
        tvpParam.rows.add(27.22); // percentage
        tvpParam.rows.add(28); // percentage
        const request = pool.request();
        request.input('tvpParam', /*sql.TVP,*/ tvpParam);
        return request.execute('[testschema].[simple.withDecimalTVP]');
    }
```

This is successfully called and the data is passed to the SQL server, but after tracking the queries (using SQL Server Profiler), this is actually what the database recieve as query:
```
declare @p3 testschema.withDecimalTypeTT
insert into @p3 values(3011)
insert into @p3 values(2722)
insert into @p3 values(28)

exec sp_executesql N'declare @___return___ int;exec @___return___ = [testschema].[simple.withDecimalTVP] @tvpParam=@P1;select @___return___ as ''___return___'';',N'@P1 [testschema].[withDecimalTypeTT] READONLY',@p3
```
As you can notice, the decimal point is lost and the digits are transformed into non-decimal. Digging deeper inside the driver code, I realised the `precision` and `scale` values for Decimal types are not passed correctly, so in [THIS COMMIT](https://github.com/milenkovyua/node-sqlserver-v8/commit/df8926b188c2b1a09a2860efb25213087cb0ca83) are the changes I made on a local branch to fix that issue inside the function `TvpFromTable`. That still doesn't solved the problem. One thing to note is, the problem occurs after adding the value of `tvpParam.rows.add(28)`, to the table. In every case we add number, without decimal part in it, the issue appears.

This is the repo I have prepared to reproduce the bug:
https://github.com/milenkovyua/msnodesqlv8test

**Comment by TimelordUK** [comment]
are you able to compile the code locally for testing

As it turns out sending an array of numeric to server via the c driver will not work unless they are scaled the same hence this problem

I have not had chance to fully test but looks the easiest way is to force the driver to use doubles rather than numerics for array binding.

i.e. in BoundDatum.cpp

```cpp
	void BoundDatum::sql_numeric(const Local<Value> pp)
	{
		if (pp->IsArray())
		{
			bind_double_array(pp);
		}
		else
		{
			bind_numeric(pp);
		}
	}
```

alternatively in the javascript code where you patched precision to build the table type, you can probably say use type double instead of numeric but I have not tested that either.

**Comment by milenkovyua** [comment]
Thanks for the respone @TimelordUK 

That worked. I used to perform tests both on the repo I prepared and in another one of our projects and it worked. There are still some differences in the TVP values inside the  constructed query, as now it look like this:
```
declare @p3 testschema.withDecimalTypeTT
insert into @p3 values(30,109999999999999)
insert into @p3 values(27,219999999999999)
insert into @p3 values(28)
```
And yet this is not something that matter so much, as at the end the values stored inside the database and the tables are correct.

This works, with no matter the patches in ( https://github.com/milenkovyua/node-sqlserver-v8/commit/df8926b188c2b1a09a2860efb25213087cb0ca83 ) and yet, I think it's good to have it there, as the transforming functions are counting on these values. Also this commit include a bugfix for DateTime2 TVP columns, which throws error if the column's first element value is null.
